### ISSUE TITLE
Custom locales

### ISSUE THREAD

**Issue by ryleyb** [houseabsolute/DateTime-Locale#19]
In older versions of this library (Debian jessie - 0.45), I made a subclass of `en_US` for some custom changes.  Looking at a newish version (Debian stretch - 1.11), that no longer seems to be possible.  

I saw the note in `DateTime::Locale::Base` saying it is depreciated, which suggests to me that subclassing is no longer the preferred way to handle this.

This is the type of thing I had before:
```perl
package DateTime::Locale::en_US_CUSTOM;
use DateTime::Locale::en_US;
use base 'DateTime::Locale::en_US';

sub time_format_full   { 'HH:mm:ss zzzz' }  
#... a few more of that type
1;                                                                                                                                           
```

What is the recommended way of handling this in newer versions?

**Comment by autarch** [comment]
This should still work. The code is all still there, it's just not documented. I honestly didn't think anyone was using it.

I think the way to address this is to make a new method like `DateTime::Locale->register` but to make it possible to do so by providing the data that is passed to `DateTime::Locale::FromData->new`. This is actually pretty straightforward, since you just have to pass a big hashref of data.

I'd be open to a PR. If you want to discuss it you might want to join the houseabsolute slack channel's #datetime. If you're interested send your email address to me at autarch@urth.org and I will send you an invite.

**Comment by ryleyb** [comment]
The nice thing about how it was working before is that I could just subclass `en_US` and change the bits I need.  It seems in the new method I'd have to fully specify my custom Locale? 

Anyways, I think the potential issue I was raising is that `DateTime::Locale::en_US` no longer exists... That's how I got here.  `DateTime::Locale::Base` exists but none of the actual locale .pm files do.

**Comment by autarch** [comment]
Ah, that's true about the per-locale modules no longer existing. We could add a method to `FromData` that returns a copy of the data hashref that was passed to the constructor. Then you could take that, make a few changes, and make a new locale from that. Combining that with some sort of new registration mechanism sounds like it would do what you want, I think.

**Comment by autarch** [comment]
Released in 1.21.
